### PR TITLE
Add cmake option AOTRITON_NAME_SUFFIX to resolve name conflicts

### PR DESCRIPTION
**Pull request by xinyazhang**
This adds suffix to namespace `aotriton` and the library file prefix `libaotriton`. For example, build with `-DAOTRITON_NAME_SUFFIX=123` will generate the library with `namespace aotrition123` and `libaotriton123_v2.so`.

This is to address potential name conflicts with PyTorch since PyTorch will bundle AOTriton shared library starting 2.4+

### CODE REVIEW

File: bindings/module.cc [hunk 1]
```diff
@@ -217,6 +221,17 @@ namespace pyaotriton {
       .def_property_readonly("strides", &aotriton::TensorView<0>::strides)
       .def_property_readonly("data_ptr", &aotriton::TensorView<0>::data_ptr)
       .def_property_readonly("dtype", &aotriton::TensorView<0>::dtype);
+    m.def("get_name_suffix",
+#if AOTRITON_ENABLE_SUFFIX
+#define xstr(s) str(s)
+#define str(s) #s
+          []() -> std::string { return xstr(AOTRITON_NAME_SUFFIX); }
```
Comment (jithunnair-amd) [inline bindings/module.cc:228]: @xinyazhang I'm sure there's some nuance with the preprocessing flow here, but can you elaborate why we need two macros `xstr` and `str`?

File: v2src/CMakeLists.txt [hunk 2]
```diff
@@ -206,8 +215,10 @@ install(FILES "${AOTRITON_V2_BUILD_DIR}/${AOTRITON_LIBRARY_FILE}" DESTINATION ${
...
-add_dependencies(aotriton aotriton_v2)
+# add_dependencies(aotriton aotriton_v2)
 # target_link_libraries(aotriton INTERFACE ${CMAKE_INSTALL_PREFIX}/lib/libaotriton_v2.a)
 # target_include_directories(aotriton INTERFACE ${CMAKE_INSTALL_PREFIX}/include)
-target_link_libraries(aotriton INTERFACE ${AOTRITON_V2_BUILD_DIR}/${AOTRITON_LIBRARY_FILE})
-target_include_directories(aotriton INTERFACE ${CMAKE_CURRENT_SOURCE_PARENT_DIR}/include)
+# target_link_libraries(aotriton INTERFACE ${AOTRITON_V2_BUILD_DIR}/${AOTRITON_LIBRARY_FILE})
+# target_include_directories(aotriton INTERFACE ${CMAKE_CURRENT_SOURCE_PARENT_DIR}/include)
+# target_include_directories(aotriton INTERFACE ${CMAKE_CURRENT_SOURCE_PARENT_DIR}/include)
+target_link_libraries(aotriton INTERFACE aotriton_v2)
```
Comment (jithunnair-amd) [inline v2src/CMakeLists.txt:224]: @xinyazhang Are these changes related to the name suffix feature? Can you please explain this changeset?

File: bindings/module.cc (hunk 1 above)
Comment (xinyazhang) [reply bindings/module.cc:228]: @jithunnair-amd https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html

File: v2src/CMakeLists.txt (hunk 2 above)
Comment (xinyazhang) [reply v2src/CMakeLists.txt:224]: the suffix is not long added to the symbols, but also added to the output files. Linking `aotriton INTERFACE` can eliminate the necessary to construct the output file name with suffix again.

**Comment by jithunnair-amd** [comment]
Build was successful and produced following outputs:
```
#ls -lart /usr//lib/aotriton/lib/
total 274288
-rw-r--r-- 1 root root 280861184 Sep 27 20:40 libaotritonmy_suffix_v2.so

#grep suffix /usr//lib/aotriton/include/aotriton/ -r
/usr//lib/aotriton/include/aotriton/config.h:#define AOTRITON_NAME_SUFFIX my_suffix
/usr//lib/aotriton/include/aotriton/config.h:#define AOTRITON_NS aotritonmy_suffix
```

**Review by jithunnair-amd — APPROVED** [review]
